Enum member after `NaN` must have initializer in tsgo, which is not in ts6.0

## Steps to reproduce

```ts
enum E {
    A = 0 / 0,
    B,
}
```

## Behavior with `typescript@6.0`

No error. `E.B` is also `NaN`.

## Behavior with `tsgo`

error TS1061: Enum member must have initializer.

--- a/internal/checker/checker.go
+++ b/internal/checker/checker.go
@@ -23768,22 +23768,23 @@ func (c *Checker) computeEnumMemberValues(node *ast.Node) {
 	nodeLinks := c.nodeLinks.Get(node)
 	if !(nodeLinks.flags&NodeCheckFlagsEnumValuesComputed != 0) {
 		nodeLinks.flags |= NodeCheckFlagsEnumValuesComputed
-		var autoValue jsnum.Number
+		autoValue := new(jsnum.Number)
 		var previous *ast.Node
 		for _, member := range node.Members() {
 			result := c.computeEnumMemberValue(member, autoValue, previous)
 			c.enumMemberLinks.Get(member).value = result
 			if value, isNumber := result.Value.(jsnum.Number); isNumber {
-				autoValue = value + 1
+				nextValue := value + 1
+				autoValue = &nextValue
 			} else {
-				autoValue = jsnum.NaN()
+				autoValue = nil
 			}
 			previous = member
 		}
 	}
 }
 
-func (c *Checker) computeEnumMemberValue(member *ast.Node, autoValue jsnum.Number, previous *ast.Node) evaluator.Result {
+func (c *Checker) computeEnumMemberValue(member *ast.Node, autoValue *jsnum.Number, previous *ast.Node) evaluator.Result {
 	if ast.IsComputedNonLiteralName(member.Name()) {
 		c.error(member.Name(), diagnostics.Computed_property_names_are_not_allowed_in_enums)
 	} else if ast.IsBigIntLiteral(member.Name()) {
@@ -23806,7 +23807,7 @@ func (c *Checker) computeEnumMemberValue(member *ast.Node, autoValue jsnum.Numbe
 	// If the member is the first member in the enum declaration, it is assigned the value zero.
 	// Otherwise, it is assigned the value of the immediately preceding member plus one, and an error
 	// occurs if the immediately preceding member is not a constant enum member.
-	if autoValue.IsNaN() {
+	if autoValue == nil {
 		c.error(member.Name(), diagnostics.Enum_member_must_have_initializer)
 		return evaluator.NewResult(nil, false, false, false)
 	}
@@ -23817,7 +23818,7 @@ func (c *Checker) computeEnumMemberValue(member *ast.Node, autoValue jsnum.Numbe
 			c.error(member.Name(), diagnostics.Enum_member_following_a_non_literal_numeric_member_must_have_an_initializer_when_isolatedModules_is_enabled)
 		}
 	}
-	return evaluator.NewResult(autoValue, false, false, false)
+	return evaluator.NewResult(*autoValue, false, false, false)
 }
 
 func (c *Checker) computeConstantEnumMemberValue(member *ast.Node) evaluator.Result {

--- a/testdata/baselines/reference/compiler/enumAutoIncrementValue.js
+++ b/testdata/baselines/reference/compiler/enumAutoIncrementValue.js
@@ -1,0 +1,16 @@
+//// [tests/cases/compiler/enumAutoIncrementValue.ts] ////
+
+//// [a.ts]
+enum E {
+    A = 0 / 0,
+    B,
+}
+
+
+//// [a.js]
+"use strict";
+var E;
+(function (E) {
+    E[E["A"] = NaN] = "A";
+    E[E["B"] = NaN] = "B";
+})(E || (E = {}));

--- a/testdata/baselines/reference/compiler/enumAutoIncrementValue.symbols
+++ b/testdata/baselines/reference/compiler/enumAutoIncrementValue.symbols
@@ -1,0 +1,13 @@
+//// [tests/cases/compiler/enumAutoIncrementValue.ts] ////
+
+=== a.ts ===
+enum E {
+>E : Symbol(E, Decl(a.ts, 0, 0))
+
+    A = 0 / 0,
+>A : Symbol(E.A, Decl(a.ts, 0, 8))
+
+    B,
+>B : Symbol(E.B, Decl(a.ts, 1, 14))
+}
+

--- a/testdata/baselines/reference/compiler/enumAutoIncrementValue.types
+++ b/testdata/baselines/reference/compiler/enumAutoIncrementValue.types
@@ -1,0 +1,16 @@
+//// [tests/cases/compiler/enumAutoIncrementValue.ts] ////
+
+=== a.ts ===
+enum E {
+>E : E
+
+    A = 0 / 0,
+>A : E.A
+>0 / 0 : number
+>0 : 0
+>0 : 0
+
+    B,
+>B : E.B
+}
+

--- a/testdata/tests/cases/compiler/enumAutoIncrementValue.ts
+++ b/testdata/tests/cases/compiler/enumAutoIncrementValue.ts
@@ -1,0 +1,5 @@
+// @filename: a.ts
+enum E {
+    A = 0 / 0,
+    B,
+}

